homeassistant/components/flexit_bacnet/climate.py

```diff
@@ -0,0 +1,177 @@
+    VENTILATION_MODES,
+    FlexitBACnet,
+)
+from flexit_bacnet.bacnet import DecodingError
+
+from homeassistant.components.climate import (
+    PRESET_AWAY,
+    PRESET_BOOST,
+    PRESET_HOME,
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    DOMAIN,
+    PRESET_TO_VENTILATION_MODE_MAP,
+    VENTILATION_TO_PRESET_MODE_MAP,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_devices: AddEntitiesCallback,
+) -> None:
+    """Set up the Flexit Nordic unit."""
+    device = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_devices([FlexitClimateEntity(device)])
+
+
+class FlexitClimateEntity(ClimateEntity):
+    """Flexit air handling unit."""
+
+    _attr_has_entity_name = True
+
+    _attr_hvac_modes = [
+        HVACMode.OFF,
+        HVACMode.FAN_ONLY,
+    ]
+
+    _attr_preset_modes = [
+        PRESET_AWAY,
+        PRESET_HOME,
+        PRESET_BOOST,
+    ]
+
+    _attr_supported_features = (
+        ClimateEntityFeature.PRESET_MODE
+        | ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.AUX_HEAT
+    )
+
+    _attr_target_temperature_step = 1.0
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+
+    def __init__(self, device: FlexitBACnet) -> None:
+        """Initialize the unit."""
+        self._device = device
+        self._attr_unique_id = device.serial_number
+        self._attr_device_info = DeviceInfo(
+            identifiers={
+                (DOMAIN, device.serial_number),
+            },
+            manufacturer="Flexit",
+            model="Nordic",
+        )
+
+    async def async_update(self) -> None:
+        """Refresh unit state."""
+        await self._device.update()
+
+    @property
+    def name(self) -> str:
+        """Name of the entity."""
+        return "Ventilation"
+
+    @property
+    def current_temperature(self) -> float:
+        """Return the current temperature."""
+        return self._device.room_temperature
+
+    @property
+    def target_temperature(self) -> float:
+        """Return the temperature we try to reach."""
+        if self._device.ventilation_mode == VENTILATION_MODES[VENTILATION_MODE_AWAY]:
+            return self._device.air_temp_setpoint_away
+
+        return self._device.air_temp_setpoint_home
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        if (temperature := kwargs.get(ATTR_TEMPERATURE)) is None:
+            return
+
+        try:
+            if (
+                self._device.ventilation_mode
```

Comment:
you do this a few times, why not create a property is_away() or is_home() or something like that so you can just call that instead of repeating this logic

Reply:
I cleaned up the ventilation modes a bit, so hopefully this isn't a big issue any more. Please let me know if it isn't good enough.